gdx/src/com/badlogic/gdx/graphics/glutils/FileTextureArrayData.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.glutils;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.GL30;
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.graphics.TextureArrayData;
+import com.badlogic.gdx.graphics.TextureData;
+import com.badlogic.gdx.utils.GdxRuntimeException;
+
+/** @author Tomski **/
+public class FileTextureArrayData implements TextureArrayData {
+
+	private TextureData[] textureDatas;
+	private boolean prepared;
+	private Pixmap.Format format;
+	private int depth;
+	boolean useMipMaps;
+
+	public FileTextureArrayData (Pixmap.Format format, boolean useMipMaps, FileHandle[] files) {
+		this.format = format;
+		this.useMipMaps = useMipMaps;
+		this.depth = files.length;
+		textureDatas = new TextureData[files.length];
+		for (int i = 0; i < files.length; i++) {
+			textureDatas[i] = TextureData.Factory.loadFromFile(files[i], format, useMipMaps);
+		}
+	}
+
+	@Override
+	public boolean isPrepared () {
+		return prepared;
+	}
+
+	@Override
+	public void prepare () {
+		int width = -1;
+		int height = -1;
+		for (TextureData data : textureDatas) {
```

Comment:
Nitpicking: I'd either use an libgdx Array or iterate over textureDatas using a `for(int i....)` loop, so we don't create (unnecessary) garbage.